AdminServer/appscale/admin/__init__.py

```diff
@@ -130,20 +132,255 @@ def wait_for_delete(operation_id, http_port):
+
+    version_node = constants.VERSION_NODE_TEMPLATE.format(
+      project_id=project_id, service_id=service_id, version_id=version_id)
+
+    yield self.thread_pool.submit(self.version_update_lock.acquire)
+    try:
+      try:
+        self.zk_client.delete(version_node)
+      except NoNodeError:
+        pass
+    finally:
+      self.version_update_lock.release()
+
+    version_key = VERSION_PATH_SEPARATOR.join([project_id, service_id,
+                                               version_id])
+    try:
+      self.acc.stop_version(version_key)
+    except AppControllerException as error:
+      message = 'Error while stopping version: {}'.format(error)
+      raise CustomHTTPError(HTTPCodes.INTERNAL_ERROR, message=message)
+
+    raise gen.Return(http_port)
+
+class ProjectsHandler(BaseVersionHandler):
+  """ Manages application. """
+
+  def initialize(self, acc, ua_client, zk_client, version_update_lock,
+                 thread_pool):
+    """ Defines required resources to handle requests.
+
+    Args:
+      acc: An AppControllerClient.
+      ua_client: A UAClient.
+      zk_client: A KazooClient.
+      version_update_lock: A kazoo lock.
+      thread_pool: A ThreadPoolExecutor.
+    """
+    self.acc = acc
+    self.ua_client = ua_client
+    self.zk_client = zk_client
+    self.version_update_lock = version_update_lock
+    self.thread_pool = thread_pool
+
+  @gen.coroutine
+  def wait_for_delete(self, ports_to_close, project_path):
+    """ Tracks the progress of removing version(s).
+  
+    Args:
+      ports_to_close: A list of integers specifying the ports to wait for.
+      project_path: The zookeeper path to the project that should be deleted 
+        after the ports are closed.
+    Raises:
+      OperationTimeout if the deadline is exceeded.
+    """
+
+    start_time = time.time()
+    deadline = start_time + constants.MAX_OPERATION_TIME
+
+    finished = 0
+    ports = ports_to_close[:]
+    while True:
+      if time.time() > deadline:
+        message = 'Delete operation took too long.'
+        raise CustomHTTPError(HTTPCodes.INTERNAL_ERROR, message=message)
+      to_remove = []
+      for http_port in ports:
+        # If the port is open, continue to process other ports.
+        if utils.port_is_open(options.login_ip, int(http_port)):
+          continue
+        # Otherwise one more port has finished and remove it from the list of
+        # ports to check.
+        finished += 1
+        to_remove.append(http_port)
+      ports = [p for p in ports if p not in to_remove]
+      if finished == len(ports_to_close):
+        break
+
+      yield gen.sleep(1)
+
+    # Cleanup the project in zookeeper.
+    yield self.thread_pool.submit(self.version_update_lock.acquire)
+    try:
+      try:
+        self.zk_client.delete(project_path, recursive=True)
+      except NoNodeError:
+        pass
+    finally:
+      self.version_update_lock.release()
+
+  @gen.coroutine
+  def post(self):
```

Comment:
This should be a GET, and the output should be a list of Project objects. https://cloud.google.com/resource-manager/reference/rest/v1/projects#Project